### PR TITLE
[fsdp] fix: use non-inplace div for activation-offloaded logits views

### PR DESCRIPTION
**Pull request by cavities12**
### What does this PR do?

Fixes a `RuntimeError` in `dp_actor.py` (legacy workers) and `engine/fsdp/transformer_impl.py` (engine workers) when using FSDP2 with `activation_offload=True`.

When activation offload is enabled, `squeeze()` is wrapped in a custom autograd Function (`SqueezeBackward1`), producing a view that cannot be modified inplace. PyTorch raises `RuntimeError` because `div_()` would corrupt the custom backward function attached to the view.

Both `div_(temperature)` call sites in `_forward_micro_batch` are affected (the `use_remove_padding` path at line 259 and the non-rmpad path at line 371).

### Checklist Before Starting

- [x] Search for similar PRs: [inplace div activation offload](https://github.com/volcengine/verl/pulls?q=is%3Apr+inplace+div+activation+offload)
- [x] Format the PR title as `[{modules}] {type}: {description}`

### Test

**Production validation:** Verified during GRPO training with FSDP2 + `activation_offload=True` on 8xH200. The `RuntimeError` no longer occurs.

**Unit test:** Added `tests/workers/actor/test_inplace_div_activation_offload.py` (CPU-only, runs with `pytest` directly — no `torchrun` or distributed setup required):

| Test | Verifies |
|------|----------|
| `test_inplace_div_fails_on_activation_offload_view` | Inplace `div_()` raises `RuntimeError` on activation-offloaded views |
| `test_non_inplace_div_works_on_activation_offload_view` | Non-inplace `/` works on such views |
| `test_conditional_div_with_grad_enabled` | Conditional logic selects non-inplace path when grad is enabled |
| `test_conditional_div_with_no_grad` | Conditional logic selects inplace path under `no_grad()` |
| `test_inplace_div_safe_on_offload_view_under_no_grad` | Inplace `div_()` is safe on offloaded views under `no_grad()` |
| `test_gradient_flows_through_non_inplace_path` | Gradients propagate correctly through the non-inplace path |

### API and Usage Example

No API changes.

### Design & Code Changes

Both `div_(temperature)` calls in `_forward_micro_batch` are replaced with a conditional:

```diff
- logits.div_(temperature)
+ if torch.is_grad_enabled():
+     logits = logits / temperature  # non-inplace for activation offload views
+ else:
+     logits.div_(temperature)
```

**Why conditional rather than always non-inplace?** `compute_log_prob` and `compute_ref_log_prob` run under `torch.no_grad()`, where activation offload is inactive and inplace `div_()` is safe. Keeping `div_()` on this path avoids allocating a second full copy of the logits tensor — which can be significant (e.g., ~5.6 GB for a 19K-token sequence with vocab_size=151936 in bf16).

`update_policy` runs with grad enabled, so it takes the non-inplace path — this is the only codepath where the `RuntimeError` occurs.

**Out of scope:** `megatron_actor.py` and `torchtitan/transformer_impl.py` contain the same `div_(temperature)` pattern. If those paths also support `activation_offload=True`, they may need the same fix.

### Checklist Before Submitting

- [x] Read the [Contribute Guide](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md).
- [x] Apply [pre-commit checks](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [x] Add / Update [the documentation](https://github.com/volcengine/verl/tree/main/docs). — N/A, no user-facing API change.
- [x] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/volcengine/verl/tree/main/.github/workflows): `tests/workers/actor/test_inplace_div_activation_offload.py` (CPU-only, runs with `pytest`).
- [x] Once your PR is ready for CI, send a message in [the `ci-request` channel](https://verl-project.slack.com/archives/C091TCESWB1). — Will request after review.
- [x] If your PR is related to the `recipe` submodule, update the reference. — N/A.

### CODE REVIEW

**Comment by wuxibin89** [comment]
legacy worker(fsdp_workers.py/megatron_workers.py) will soon be deprecated in v0.7.2, please use `engine_workers.py` by
```
trainer.use_legacy_worker_impl=disable
```

**Comment by cavities12** [comment]
@wuxibin89 ported the fix!

**Comment by wuxibin89** [comment]
@cavities12 Please run pre-commit before submit PR.
```
pip install pre-commit
pre-commit install
pre-commit run
```

**Comment by cavities12** [comment]
@wuxibin89 fixed, sorry! I did run precommit but turns out my ruff version was different from yours
